### PR TITLE
82 Allow use for full URL for consistency check

### PR DESCRIPTION
**Pull request by gwshaw**

### CODE REVIEW

File: lib/apivore/rspec_matchers.rb [hunk 1]
```diff
@@ -22,8 +22,18 @@ module RspecMatchers
...
-        JSON.parse(req)
+        res =
+          if master_swagger_url.starts_with? 'http'
+            Net::HTTP.get_response(URI(master_swagger_url))
+          else
+            Net::HTTP.get_response(master_swagger_url, "/swagger.json")
+          end
+
+        unless res.is_a? Net::HTTPSuccess
+          message = "Master swagger at #{master_swagger_url} not accessible\n"
```
Comment (emirkin) [inline lib/apivore/rspec_matchers.rb]: Thanks for this! It was frustrating to see unrelated error messages when swagger download was failing.

File: lib/apivore/rspec_matchers.rb [hunk 2]
```diff
@@ -22,8 +22,18 @@ module RspecMatchers
       end
 
       define_method :fetch_master_swagger do
-        req = Net::HTTP.get(master_swagger_host, "/swagger.json")
-        JSON.parse(req)
+        res =
+          if master_swagger_url.starts_with? 'http'
```
Comment (emirkin) [inline lib/apivore/rspec_matchers.rb]: What is the reasoning behind this logic?

Comment (gwshaw) [inline lib/apivore/rspec_matchers.rb]: `Net::HTTP.get` takes either a host string and path parameters or a full URI. One distinguishing feature of a full URI is that the protocol is specified. This logic is simple and adequate to allow http or https.

**Comment by luizvarela** [comment]
LGTM

**Pull request merged**